src/View/Widgets/HeaderBar.vala

```diff
@@ -266,6 +266,32 @@ public class Files.View.Chrome.HeaderBar : Hdy.HeaderBar {
         });
 
         location_bar.escape.connect (() => {escape ();});
+
+        undo_manager = UndoManager.instance ();
+        undo_manager.request_menu_update.connect (set_undo_redo_tooltips);
+        set_undo_redo_tooltips ();
+    }
+
+    private void set_undo_redo_tooltips () {
+        var undo_action_s = undo_manager.get_next_undo_description ();
+        var redo_action_s = undo_manager.get_next_redo_description ();
+        if (undo_action_s != "") {
+            undo_button.tooltip_markup = Granite.markup_accel_tooltip (
+                undo_accels,
+                _("Undo %s").printf (undo_action_s)
+            );
+        } else {
+            undo_button.tooltip_text = _("No operation to undo");
```

Comment:
I think we should still show the accel here since it will still be useful to teach about its existence